zds/tutorialv2/tests/tests_views/tests_addcontributor.py

```diff
@@ -61,26 +61,14 @@ def test_authenticated_author(self):
         response = self.client.post(self.form_url, self.form_data)
         self.assertRedirects(response, self.content_url)
 
-    def test_authenticated_staff_tutorial(self):
+    def test_authenticated_staff(self):
         self.client.force_login(self.staff)
-        self.content.type = "TUTORIAL"
-        self.content.save()
-        response = self.client.post(self.form_url, self.form_data)
-        self.assertRedirects(response, self.content_url)
-
-    def test_authenticated_staff_article(self):
-        self.client.force_login(self.staff)
-        self.content.type = "ARTICLE"
-        self.content.save()
-        response = self.client.post(self.form_url, self.form_data)
-        self.assertRedirects(response, self.content_url)
-
-    def test_authenticated_staff_opinion(self):
-        self.client.force_login(self.staff)
-        self.content.type = "OPINION"
-        self.content.save()
-        response = self.client.post(self.form_url, self.form_data)
-        self.assertEqual(response.status_code, 403)
+        types = ["TUTORIAL", "ARTICLE", "OPINION"]
+        for type in types:
+            self.content.type = type
+            self.content.save()
+            response = self.client.post(self.form_url, self.form_data)
+            self.assertRedirects(response, self.content_url)
```

Comment:
```suggestion
        for type in CONTENT_TYPE_LIST:
            with self.subTest(type):
                self.content.type = type
                self.content.save()
                response = self.client.post(self.form_url, self.form_data)
                self.assertRedirects(response, self.content_url)
```
En rajoutant
```py
from zds.tutorialv2.models import CONTENT_TYPE_LIST
```
Comme ça, le jour où on rajoute un autre type de contenu, ce sera transparent. Le `subTest` permet de voir plus facilement quel cas fait planter le test.

Reply:
Bonne idée, c'est fait;